internal/manifest/list.go

```diff
@@ -68,6 +68,7 @@ type ListUpdate struct {
 	Digest    digest.Digest
 	Size      int64
 	MediaType string
+	Platform  imgspecv1.Platform
```

Comment:
> includes the fields _which a List's UpdateInstances() method will modify_.

I don’t see any code added to deal with modification.

That’s… probably just as well, because adding a field that will cause `Platform` to be cleared by older callers of `.UpdateInstance` would be a compatibility break.

But we need to somehow deal with _both_ the need to add the new feature, and the requirement not to break any existing API caller.

---

One clean way would be to not change the `ListPublic` interface at all, and expose all of this in a new way.

`c/image/manifest.ListGetInstanceInfo(List[Public], digest.Digest) (manifest.SomeNewStructWithNewFields)`

OTOH that’s tedious to implement.

Probably simpler would be to do the what this PR does, just add a new field to the struct, but somehow clearly demarcate that it is inspect-only. `ListUpdate.ReadOnly.Platform` ?? That’s fairly ugly but plausible.

Reply:
Oops… I’m sorry, for some reason I thought the intended caller is external, Podman or so.

If this is for the internal c/image/copy caller, we don’t need to change `ListPublic` at all: add a new `List.Convert` (just like `ConvertToMIMEType`, but returns at the internal type), and a new `List.InstanceData()` returning any fields we like.

… but again, that’s tedious and just adding a “read-only” field to the public API would be simpler to write _code_ for. At the cost of having to write a much more careful documentation… because any pre-existing external callers must not be broken, and any pre-existing external implementers of `.Instance()` must not be broken either.

Reply:
> … but again, that’s tedious and just adding a “read-only” field to the public API would be simpler to write code for. At the cost of having to write a much more careful documentation… because any pre-existing external callers must not be broken, and any pre-existing external implementers of .Instance() must not be broken either.

I have made this field private with a comment of `read-only` and created a public Get API for this.

Reply:
I don’t see that a method on `ListUpdate` can help here:
- Anyone can call the added method (not just packages allowed to import `c/image/internal/manifest`). So this does not hide the read side.
- OTOH the external callers can’t set the value. So to the extent we need to worry about external implementations of `List`, they are now worse off because they can’t keep up with callers.
- The “fields which a List's UpdateInstances() method will modify.” comment is still incorrect.

Reply:
I agree, how about just adding a comment to the public field itself ? Did you mean this in https://github.com/containers/image/pull/2029#discussion_r1255165714

Reply:
Yes, I think just documenting that is practical enough…